docs/dev/directory_structure.rst

```diff
@@ -0,0 +1,30 @@
+Directory Structure
+===================
+
+This is not intended to be an exhaustive list, just a summary with some comments, to help new devs find things.  Starting in the root of the repository:
+
+* ``docs``: Fairly new and therefore very incomplete; add to it!
+
+  * ``admin``
+  * ``dev``
+
+* ``esp``: The root of the django project.
+
+  * ``esp``: The root of the python code.  Every folder from here on down needs to have an ``__init__.py``.  Contains a directory for each django app we use, along with some other files and directories.  There are a whole lot of these, but the most important are below.  Each app contains a file or directory for models, views, and tests; most also contain a directory for migrations.
+
+    * ``accounting``: The website’s accounting system (``accounting_core`` and ``accounting_docs`` are old and deprecated).
+    * ``program``: This app contains most of the site’s logic.  For the most part it’s a fairly normal Django app, but it also contains all the program modules.
+
+      * ``models``: The main models for ``Program``, ``ClassSubject``, ``ClassSection``; many of the other models used by programs live in other apps, such as ``resources``.
+      * ``modules``: This directory contains all the program modules.  The python code for the individual modules is in ``handlers/<name>module.py``; most of the remainder of the directory won’t be relevant to you.  See `<program_modules.rst>`_ for details on the program module system.
+
+    * ``qsd``: This app contains all of the code for QSD (which is the website’s system for admin-editable "quasi-static" text).
+    * ``users``: This app contains important models like ``ESPUser``, ``Permission``, ``Record``, ``ContactInfo``, and so on.
+    * ``*_settings.py``: Various files in which our django settings are kept.  ``django_settings.py`` contains settings used by all sites; ``local_settings.py`` is generated by the setup script and contains site-specific settings.
+    * ``urls.py``: Master URL routing table.
+
+  * ``manage.py``: Runs django commands like ``./manage.py shell_plus``.
+  * ``public``: Served statically from the root of the website.  (So ``/esp/public/media/scripts/foo.js`` would be served at ``esp.mit.edu/media/scripts/foo.js``.)
```

Comment:
I don't think this is accurate. ``public/media`` is served statically at ``/media``, and that's it. If you added another directory to ``public``, it wouldn't be served.

Also, can you use a generic hostname here?